File: go/etl/etl.go
package etl

import (
	"strings"
)

// Transform converts old map format to new map format
func Transform(in map[int][]string) map[string]int {
	out := make(map[string]int)
	for k, v := range in {
		for _, w := range v {
			out[strings.ToLower(w)] = k
		}
	}
	return out
}

File: cpp/etl/etl.cpp
#include "etl.h"

namespace etl {

    newmap transform(const oldmap &old) {
        newmap result;
        for (const auto &mit : old) {
            const auto points = mit.first;
            const auto chars = mit.second;
            for (const auto &vit : chars) {
                result[tolower(vit)] = points;
            }
        }
        return result;
    }

}  // namespace etl
